azure/services/privateendpoints/spec_test.go

```diff
@@ -20,332 +20,228 @@ import (
-							GroupIDs:             nil,
-							RequestMessage:       ptr.To(privateEndpoint1.PrivateLinkServiceConnections[0].RequestMessage),
-						},
-					}},
-					ManualPrivateLinkServiceConnections: []*armnetwork.PrivateLinkServiceConnection{},
-					ProvisioningState:                   ptr.To(armnetwork.ProvisioningStateSucceeded),
 				},
-				Tags: map[string]*string{"sigs.k8s.io_cluster-api-provider-azure_cluster_my-cluster": ptr.To("owned"), "Name": ptr.To("test-private-endpoint1")},
 			},
-			expect: func(g *WithT, result interface{}) {
-				g.Expect(result).To(BeNil())
+			AzureName: fakePrivateEndpoint.Name,
+			PrivateLinkServiceConnections: []asonetworkv1.PrivateLinkServiceConnection{{
+				Name: ptr.To(fakePrivateEndpoint.PrivateLinkServiceConnections[0].Name),
+				PrivateLinkServiceReference: &genruntime.ResourceReference{
+					ARMID: fakePrivateEndpoint.PrivateLinkServiceConnections[0].PrivateLinkServiceID,
+				},
+				GroupIds:       fakePrivateEndpoint.PrivateLinkServiceConnections[0].GroupIDs,
+				RequestMessage: ptr.To(fakePrivateEndpoint.PrivateLinkServiceConnections[0].RequestMessage),
+			}},
+			Owner: &genruntime.KnownResourceReference{
+				Name: fakePrivateEndpoint.ResourceGroup,
+			},
+			Subnet: &asonetworkv1.Subnet_PrivateEndpoint_SubResourceEmbedded{
+				Reference: &genruntime.ResourceReference{
+					ARMID: fakePrivateEndpoint.SubnetID,
+				},
 			},
+			IpConfigurations: []asonetworkv1.PrivateEndpointIPConfiguration{
+				{
+					PrivateIPAddress: ptr.To(fakePrivateEndpoint.PrivateIPAddresses[0]),
+				},
+				{
+					PrivateIPAddress: ptr.To(fakePrivateEndpoint.PrivateIPAddresses[1]),
+				},
+			},
+			CustomNetworkInterfaceName: ptr.To(fakePrivateEndpoint.CustomNetworkInterfaceName),
+			Location:                   ptr.To(fakePrivateEndpoint.Location),
+			Tags:                       map[string]string{"sigs.k8s.io_cluster-api-provider-azure_cluster_test_cluster": "owned", "Name": "test_private_endpoint_1", "test_tag1": "test_value1", "test_tag2": "test_value2"},
 		},
-		{
-			name: "PrivateEndpoint without AppplicationSecurityGroups already exists with the same config",
-			spec: &PrivateEndpointSpec{
-				Name:                      privateEndpoint1.Name,
-				ResourceGroup:             "test-group",
-				ClusterName:               "my-cluster",
-				ApplicationSecurityGroups: nil,
-				PrivateLinkServiceConnections: []PrivateLinkServiceConnection{{
-					Name:                 privateEndpoint1.PrivateLinkServiceConnections[0].Name,
-					GroupIDs:             privateEndpoint1.PrivateLinkServiceConnections[0].GroupIDs,
-					PrivateLinkServiceID: privateEndpoint1.PrivateLinkServiceConnections[0].PrivateLinkServiceID,
-					RequestMessage:       privateEndpoint1.PrivateLinkServiceConnections[0].RequestMessage,
-				}},
-				SubnetID: "test-subnet",
+	}
+
+	fakeASOPrivateEndpointSpecWithExtendedLocation = asonetworkv1.PrivateEndpoint_Spec{
```

Comment:
nit to reduce the amount of repeated code: it could help to use the `changes` pattern here since these PrivateEndpoint specs are so large, although it may not be worth the effort if we're only defining two private endpoints. An example is this: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/controllers/asosecret_controller_test.go#L320

Reply:
Good point. I think I can reduce at least a couple of lines. Making changes.
/hold

Reply:
Updated. Please take a look. 
/unhold